tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_Analytic.cpp

```diff
@@ -0,0 +1,199 @@
+  if constexpr (Dim == 1) {
+    Affine affine_map{-1.0, 1.0, 2.0, 3.0};
+    return domain::make_coordinate_map<Frame::ElementLogical, Frame::Inertial>(
+        affine_map);
+  } else if constexpr (Dim == 2) {
+    Affine affine_map{-1.0, 1.0, 2.0, 3.0};
+    Affine2D product_map{affine_map, affine_map};
+    return domain::make_coordinate_map<Frame::ElementLogical, Frame::Inertial>(
+        product_map);
+  } else {
+    Affine affine_map{-1.0, 1.0, 2.0, 3.0};
+    Affine3D product_map{affine_map, affine_map, affine_map};
+    return domain::make_coordinate_map<Frame::ElementLogical, Frame::Inertial>(
+        product_map);
+  }
+}
+
+template <size_t Dim>
+struct Metavariables {
+  struct factory_creation
+      : tt::ConformsTo<Options::protocols::FactoryCreation> {
+    using factory_classes = tmpl::map<
+        tmpl::pair<GeneralizedHarmonic::gauges::GaugeCondition,
+                   tmpl::list<GeneralizedHarmonic::gauges::Analytic>>,
+        tmpl::pair<evolution::initial_data::InitialData,
+                   tmpl::list<GeneralizedHarmonic::Solutions::WrappedGr<
+                                  gr::Solutions::KerrSchild>,
+                              GeneralizedHarmonic::Solutions::WrappedGr<
+                                  gr::Solutions::GaugeWave<Dim>>>>>;
+  };
+};
+
+template <size_t Dim>
+void test(const Mesh<Dim>& mesh) {
+  const auto gauge_condition = serialize_and_deserialize(
+      TestHelpers::test_creation<
+          std::unique_ptr<GeneralizedHarmonic::gauges::GaugeCondition>,
+          Metavariables<Dim>>("Analytic:\n"
+                              "  AnalyticPrescription:\n"
+                              "    GaugeWave:\n"
+                              "      Amplitude: 0.0012\n"
+                              "      Wavelength: 1.4\n")
+          ->get_clone());
+
+  const size_t num_points = mesh.number_of_grid_points();
+
+  const double time = 1.2;
+  const auto coord_map = make_coord_map<Dim>();
+  const auto logical_coords = logical_coordinates(mesh);
+  const tnsr::I<DataVector, Dim, Frame::Inertial> inertial_coords =
+      coord_map(logical_coords);
+  const auto inverse_jacobian = coord_map.inv_jacobian(logical_coords);
+
+  tnsr::a<DataVector, Dim, Frame::Inertial> gauge_h(num_points);
+  tnsr::ab<DataVector, Dim, Frame::Inertial> d4_gauge_h(num_points);
+  dynamic_cast<const GeneralizedHarmonic::gauges::Analytic&>(*gauge_condition)
+      .gauge_and_spacetime_derivative(make_not_null(&gauge_h),
+                                      make_not_null(&d4_gauge_h), mesh, time,
+                                      inertial_coords, inverse_jacobian);
+
+  CHECK_ITERABLE_APPROX(
+      gauge_h, (tnsr::a<DataVector, Dim, Frame::Inertial>(num_points, 0.0)));
+  CHECK_ITERABLE_APPROX(d4_gauge_h, (tnsr::ab<DataVector, Dim, Frame::Inertial>(
+                                        num_points, 0.0)));
+}
+
+void test_ks(const Mesh<3>& mesh) {
```

Comment:
Not a change request, but this is basically just copying the internals of the analytic gauge condition. Not really sure what the best way is to test it though. What'd be optimal is if you calculated $\Gamma_a$ by hand given some special choice of mass and spin that weren't zero and just compared the new function against what you got by hand rather than redoing the calculation. But that's a lot of work and is probably unnecessary.

Reply:
Yea, I can't think of anything better offhand because even testing the 1-index constraint requires recomputing the stuff inside the original class...